tools/clang/test/HLSLFileCheck/hlsl/diagnostics/errors/write_const_arrays.hlsl

```diff
@@ -0,0 +1,61 @@
+// RUN: %dxc -T cs_6_0 _HV 2021 %s | FileCheck %s
+
+// Array subscripts lost their qualifiers including const due to taking a different
+// path for HLSL on account of having no array to ptr decay
+// This test verifies that subscripts of constant and cbuf arrays can't be assigned
+// or otherwise altered by a few mechanisms
+
+StructuredBuffer<uint> g_robuf;
+Texture1D<uint> g_tex;
+uint g_cbuf[4];
+
+const groupshared uint gs_val[4];
+
+[NumThreads(1, 1, 1)]
+void main() {
+  const uint local[4];
+
+  //CHECK: error: read-only variable is not assignable
+  //CHECK: error: read-only variable is not assignable
+  //CHECK: error: read-only variable is not assignable
+  //CHECK: error: cannot assign to return value
+  // Assigning using assignment operator
+  local[0] = 0;
+  gs_val[0] = 0;
+  g_cbuf[0] = 0;
+  g_robuf[0] = 0;
+
+  // Assigning using out param of builtin function
+  //CHECK: error: no matching function for call to 'sincos'
+  //CHECK: error: no matching function for call to 'sincos'
+  //CHECK: error: no matching function for call to 'sincos'
+  //CHECK: error: no matching function for call to 'sincos'
+  sincos(0.0, local[0], local[0]);
```

Comment:
But is it because it's `const` or because it's `uint`?  Maybe try with `const float`?  I suppose subsequent tests with GetDimensions demonstrate that `const` at least produces the same error (as unhelpful as it is).

Reply:
Actually, this error is just fine. The note below the error line makes it clear that a const param isn't acceptable. Prior to the qualifier propagation that I added here, we got crashes.

Reply:
I wasn't aware of the note when I made the comment.  Sounds good.